src/Signup_Command.php

```diff
@@ -0,0 +1,272 @@
+	 *
+	 * @subcommand list
+	 *
+	 * @package wp-cli
+	 */
+	public function list_( $args, $assoc_args ) {
+		global $wpdb;
+
+		if ( isset( $assoc_args['fields'] ) ) {
+			$assoc_args['fields'] = explode( ',', $assoc_args['fields'] );
+		} else {
+			$assoc_args['fields'] = $this->obj_fields;
+		}
+
+		$signups = array();
+
+		$results = $wpdb->get_results( "SELECT * FROM $wpdb->signups", ARRAY_A );
+
+		if ( $results ) {
+			foreach ( $results as $item ) {
+				// Support features like --active=0.
+				foreach ( array_keys( $item ) as $field ) {
+					if ( isset( $assoc_args[ $field ] ) && $assoc_args[ $field ] !== $item[ $field ] ) {
+						continue 2;
+					}
+				}
+
+				$signups[] = $item;
+			}
+		}
+
+		$format = Utils\get_flag_value( $assoc_args, 'format', 'table' );
+
+		$formatter = $this->get_formatter( $assoc_args );
+
+		if ( 'ids' === $format ) {
+			WP_CLI::line( implode( ' ', wp_list_pluck( $signups, 'signup_id' ) ) );
+		} else {
+			$formatter->display_items( $signups );
+		}
+	}
+
+	/**
+	 * Gets details about a signup.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <signup>
+	 * : The signup ID, user login, user email, or activation key.
+	 *
+	 * [--field=<field>]
+	 * : Instead of returning the whole signup, returns the value of a single field.
+	 *
+	 * [--fields=<fields>]
+	 * : Limit the output to specific fields. Defaults to all fields.
+	 *
+	 * [--format=<format>]
+	 * : Render output in a particular format.
+	 * ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - csv
+	 *   - json
+	 *   - yaml
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Get signup.
+	 *     $ wp user signup get 1 --field=user_login
+	 *     bobuser
+	 *
+	 *     # Get signup and export to JSON file.
+	 *     $ wp user signup get bobuser --format=json > bobuser.json
+	 *
+	 * @package wp-cli
+	 */
+	public function get( $args, $assoc_args ) {
+		$signup = $this->fetcher->get_check( $args[0] );
+
+		if ( empty( $assoc_args['fields'] ) ) {
+			$assoc_args['fields'] = array_keys( (array) $signup );
+		}
+
+		$formatter = $this->get_formatter( $assoc_args );
+
+		$formatter->display_items( array( $signup ) );
+	}
+
+	/**
+	 * Activates one or more signups.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <signup>...
+	 * : The signup ID, user login, user email, or activation key of the signup(s) to activate.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Activate signup.
+	 *     $ wp user signup activate 2
+	 *     Success: Signup 2 activated. Password: bZFSGsfzb9xs
+	 *
+	 * @package wp-cli
+	 */
+	public function activate( $args, $assoc_args ) {
+		$signups = $this->fetcher->get_many( $args );
+
+		foreach ( $signups as $signup ) {
+			$result = wpmu_activate_signup( $signup->activation_key );
+
+			if ( is_wp_error( $result ) ) {
+				WP_CLI::warning( "Failed activating signup {$signup->signup_id}." );
+			} else {
+				WP_CLI::success( "Signup {$signup->signup_id} activated. Password: {$result['password']}" );
+			}
+		}
```

Comment:
We should use `WP_CLI\Utils\report_batch_operation_results()` here. Here's some example usage: https://github.com/wp-cli/extension-command/blob/2cebfa2bf786491ebf707eee0388471e885bb849/src/Theme_Command.php#L818